include/zephyr/devicetree/fixed-partitions.h

```diff
@@ -78,14 +78,12 @@ extern "C" {
 #define DT_FIXED_PARTITION_ID(node_id) DT_CAT(node_id, _PARTITION_ID)
 
 /**
- * @brief Get the node identifier of the flash memory for a partition
+ * @brief Get the node identifier of the memory for a partition
  * @param node_id node identifier for a fixed-partitions child node
  * @return the node identifier of the internal memory that contains the
- * fixed-partitions node, or @ref DT_INVALID_NODE if it doesn't exist.
+ * fixed-partitions node
  */
-#define DT_MEM_FROM_FIXED_PARTITION(node_id)                                                       \
-	COND_CODE_1(DT_NODE_HAS_COMPAT(DT_GPARENT(node_id), soc_nv_flash), (DT_GPARENT(node_id)),  \
-		    (DT_INVALID_NODE))
+#define DT_MEM_FROM_FIXED_PARTITION(node_id) DT_GPARENT(node_id)
```

Comment:
What if this points to a non-memory-mapped flash node, like a `jedec,spi-nor`?

Also, `DT_MTD_FROM_FIXED_PARTITION` would need to be adjusted. For `nordic,mram`, it can be equal to the `DT_MEM`, but not its parent (which in the case of `soc-nv-flash` would be a memory controller).

Reply:
I'm not that familiar with the fixed-partitions API extension, so I'm open to suggestions here